tools/ci_build/github/windows/post_code_coverage_to_dashboard.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+
+# command line arguments
+# --report_url=<string>
+# --report_file=<string, local file path>
+# --commit_hash=<string, full git commit hash>
+
+import argparse
+import mysql.connector
+import xml.etree.ElementTree as ET
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="ONNXRuntime test coverge report uploader for dashboard")
+    parser.add_argument("--report_url", help="URL to the Cobertura XML report")
+    parser.add_argument("--report_file", help="Path to the local cobertura XML report")
+    parser.add_argument("--commit_hash", help="Full Git commit hash")
+    return parser.parse_args()
+
+def parse_xml_report(report_file):
+    tree = ET.parse(report_file) # may throw exception
+    root = tree.getroot()
+    result = {}
+    
+    result['coverage'] = float(root.get('line-rate'))
+    result['lines_covered'] = int(root.get('lines-covered'))
+    result['lines_valid'] = int(root.get('lines-valid'))
+    return result
+
+def write_to_db(coverage_data, args):
+    # connect to database
+    cnx = mysql.connector.connect(
+        user='ort@onnxruntimedashboard', 
+        password='U5nNR7\e=UPk#Yp!', 
```

Comment:
Please don't post the password here!

Reply:
Password will have to be changed as well given this is a public PR

---
In reply to: [292754339](https://github.com/microsoft/onnxruntime/pull/1209#discussion_r292754339) [](ancestors = 292754339)

Reply:
Had a discussion on this with Changming before. The option I took on this issue was to have the particular user-account to have very restricted access (ony read/insert/delete/update rows on the particular table), while keeping the account virtually public. I'll look into other ways to obtain the password programmatically, given that the connect() API needs a password.

Reply:
Actually, given the connect() API requires a password, even if I find a way to obtain the password from some hidden vault, a developer with malicious intent can simply checkout and modify the code to dump the password, then execute it once.

Reply:
Use the secrets from Azure Keyvault, shich provides some layer of obscuration and protection. The potential access to the password is now limited to those who have branch creation access to the repo, and access to queue build job on the azure pipeline.